app/models/product_import/entry_validator.rb

```diff
@@ -312,6 +316,12 @@ def inventory_validation(entry)
     end
 
     def entry_matches_existing_variant?(entry, existing_variant)
+      # matching on the unscaled unit so we know if we are trying to update an existing variant
+      display_name_are_the_same?(entry, existing_variant) &&
+        existing_variant.unit_value == entry.unscaled_units.to_f
+    end
```

Comment:
I'm probably missing something but this seems unrelated to the refactor. Is this a bug fix?

Reply:
It looks like it, I can't really remember. I'll make sure it's in the notes to test product import